File: 4_Lists/javascript/demo_node.js
import Node from './NodeModule.js';

const node = new Node('a');

node.next = new Node('b');
node.next.next = new Node('c');

let currentNode = node;
let count = 0;
while (currentNode) {
  console.log(currentNode.data);
  count += 1;
  currentNode = currentNode.next;
}

console.log(count);

File: 4_Lists/python/demo_node.py
from NodeModule import Node

node = Node("a")

# print(node.data)

# print(node)

node.next = Node("b")

node.next.next = Node("c")

# print(node.next.data)
# print(node.next.next.data)
# print(node.next.next.next)

current_node = node
count = 0
while current_node:
    print(current_node.data)
    count = count + 1
    current_node = current_node.next

print(count)

builtin_list = ["a","b","c"]
builtin_list.append("d")
print(builtin_list)

# head - first node
# find the end of the linked list of nodes
# set next property of last node = to the append(value)

print(dir(builtin_list))




